Modules/System Tests/AL Rest Client/src/ALHttpContentTests.Codeunit.al

```diff
@@ -0,0 +1,606 @@
+codeunit 134970 "AL Http Content Tests"
+{
+    Subtype = Test;
+
+    var
+        Assert: Codeunit "Library Assert";
+
+    [Test]
+    procedure TestCreateWithText()
+    var
+        ALHttpContent: Codeunit "AL Http Content";
+        HttpContent: HttpContent;
+        HttpHeaders: HttpHeaders;
+        HeaderValues: List of [Text];
+        TextContent: Text;
+    begin
+        // [SCENARIO] Create AL Http Content object with text
+
+        // [GIVEN] AL Http Content created with text
+        ALHttpContent := ALHttpContent.Create('Hello World');
```

Comment:
Could this constructor pattern (Fluent-Interface or method-chaining pattern) become a footgun for someone not familiar with it?
This example compiles fine, but the TextContent doesn't contain the text:
```al
ALHttpContent.Create('Hello World');
TextContent := ALHttpContent.AsText();
```
Maybe it's good to mention that the return value should overwrite the caller in the summary of procedures like ALHttpContent.Create, or (if possible) if the HttpContent is by-reference you could just make the returned codeunit point to the same HttpContent instance via GetHttpContent/SetHttpContent? The third alternative would be the CurrCodeunit hack.

Reply:
Yes, I'm in favor of supporting both, the example you showed was exactly what I had in mind.